### DISCUSSION TITLE
how to import technetium blocklists?

### DISCUSSION THREAD

**Discussion by zs311521** [TechnitiumSoftware/DnsServer#1048] [Q&A] [answered]
Hi, I took a backup, but I cant seem to figure out how to import the blocklists listed by URL in the blocklist url page?

**Comment by ShreyasZare** [comment, marked as answer]
If you mean that you use the Backup Settings option to backup and want to restore it then use the Restore Backup option with the same backup zip file. You can just select DNS Settings option to restore and it will just restore the settings including the block list URLs.

**Comment by zs311521** [comment]
Thank you very much @ShreyasZare, following that guide in https://dnscrypt.info/stamps/, If I select DOH and fill out the info + generate the hash from my server, could it work, or does this require additional configuration from the application side (development). If so, is it a large endeavour? This will eliminate the need for a local technitium especially if queries can be forwarded over DOH  by the router directly to an upstream DNS resolver (technitium), with Adblocks configured on the upstream. The router will perform local caching via dnsmasq (I think).

**Reply by ShreyasZare** [reply]
For the upstream DNS server, you just need to have some TLS certificate installed to run DoH service. You can then generate a DNS stamp and it should work with any DoH client that supports it.

**Reply by zs311521** [reply]
Thanks, I generated a hash through

echo | openssl s_client -connect website.com:443 | openssl x509 -outform DER > server_cert.der

and then sha256sum server_cert.der

Then went to https://dnscrypt.info/stamps/

Selected DOH protocol, added ipv4 and ipv6 (comma separated), hostname as website.com, hash as the 256sum generated from the above query, and path as /dns-query 

I then put the hash in, but it doesn't seem to work. Did I put in the details correctly? Thank you.

**Reply by ShreyasZare** [reply]
I do not have experience using it but what I read about it, its just a thing at the DoH client side and does not need for the DoH server to have any support for it.

Check any error logs in the DoH client that you are using and see if it helps to debug it.

**Reply by zs311521** [reply]
Thank you kindly.

**Reply by zs311521** [reply]
> I do not have experience using it but what I read about it, its just a thing at the DoH client side and does not need for the DoH server to have any support for it.
> 
> 
> 
> Check any error logs in the DoH client that you are using and see if it helps to debug it.

Ah, I found the error in technitium. It suddenly stopped pinging. Do you know what it means? I renew my cert through certbot dns. 

[Timestamp] [IP_ADDRESS] System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception.
 ---> Interop+OpenSsl+SslException: SSL Handshake failed with OpenSSL error - SSL_ERROR_SSL.
 ---> Interop+Crypto+OpenSslCryptographicException: error:0A000460:SSL routines::reason(1120)
   --- End of inner exception stack trace ---
   at Interop.OpenSsl.DoSslHandshake(SafeSslHandle context, Re

**Reply by ShreyasZare** [reply]
Thanks for the details. It seems, based on the reason 1120 (SSL_R_TLSV1_ALERT_NO_APPLICATION_PROTOCOL), that the client is sending APLN in header which is not recognized or is missing. The client here is your router right?
